.github/workflows/test-spm.yml

```diff
@@ -0,0 +1,78 @@
+name: Test SPM
+
+on:
+  push:
+    branches: stable
+    paths:
+      - '.github/workflows/test-spm.yml'
+      - '**/*.swift'
+  pull_request:
+    paths:
+      - '.github/workflows/test-spm.yml'
+      - '**/*.swift'
+
+env:
+  xcode-version: 12.0.0
+
+jobs:
+  linux:
+    name: Linux
+    runs-on: ubuntu-latest
+    container: swiftgen/swift:5.3
+    steps:
+      - 
+        name: Checkout
+        uses: actions/checkout@v2
+      - 
+        name: Cache gems
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
```

Comment:
Any reason why we use this cache logic here, but on other workflows we just use `ruby/setup-ruby@v1` with `bundler-cache: true`? Why the diff?

Reply:
Because unfortunately the `ruby/setup-ruby` step/action only works on the default ubuntu images GitHub provides (and it is extremely fast on those). 

This job runs on our custom swift docker images, which don't have that, and the action is broken there.

Reply:
Gotcha. Might be worth a YAML comment to not wonder the same thing again later